### DISCUSSION TITLE
Can I put the model in simnibs for further processing in vtk? Why does the model I open have no scalar field data?

### DISCUSSION THREAD

**Discussion by clinerb** [simnibs/simnibs#322] [7.Creating head models]
There is no color distinction in the vtk model after opening, and the same is true for slices.In other words, how can I get the 3D field distribution map like in Gmsh?
![20240425130732](https://github.com/simnibs/simnibs/assets/102652655/209ae32c-ac8a-4f41-9322-52d2d57c5788)
![20240425131007](https://github.com/simnibs/simnibs/assets/102652655/1b5a3e28-78b2-4b03-9361-32d02a889107)
![20240425131106](https://github.com/simnibs/simnibs/assets/102652655/519a630d-4262-4eba-a0ae-5a55f1feea83)
thank you so much

**Comment by TorgeW** [comment]
How are you creating the vtk files? Are you using gmsh for this? If that is the case then please refer to the gmsh documentation and support :)

**Reply by clinerb** [reply]
yes,I export it from gmsh.Could you please send me a link to gmsh documentation and support

**Reply by TorgeW** [reply]
https://gmsh.info/#Documentation
https://gitlab.onelab.info/gmsh/gmsh/-/issues

**Comment by tletut0ur** [comment]
Hello, 
If anyone else is looking for simple export of simnibs results to VTK. I was able to find a simple solution with [pyvista](https://github.com/pyvista/pyvista) (full disclosure: I am a contributor). This needs to be run on the `simnibs_python` interpreter which can be found at `~/SimNIBS-{VERSION}/bin/simnibs_python` and requires pyvista to be installed on it ( activate the environment in `~/SimNIBS-{VERSION}/simnibs_env` and just `pip install pyvista`)


```python
import pyvista as pv
from pyvista import CellType


def simnibs2vtk(input_file, output_file):
    gm_surf = simnibs.read_msh(input_file)
    tets_elements_mask = gm_surf.elm.node_number_list[:, 3] != -1
    connectivity_tets = gm_surf.elm.node_number_list[tets_elements_mask] - 1 # numbering of mesh starts at 1
    connectivity_tets = np.hstack((np.full((connectivity_tets.shape[0], 1), fill_value=4), connectivity_tets))
    mesh = pv.UnstructuredGrid(connectivity_tets, 
                            np.full((connectivity_tets.shape[0]), fill_value=CellType.TETRA), 
                            gm_surf.nodes.node_coord)
    mesh.cell_data["mat"] = gm_surf.elm.tag1[tets_elements_mask]
    for data in gm_surf.elmdata:
        mesh.cell_data[data.field_name] = data.value[tets_elements_mask]
    for data in gm_surf.nodedata:
        mesh.point_data[data.field_name] = data.value
    mesh.save(output_file)
```

This exports the materials for thresholding and the attached scalars.

**Comment by jduemose** [comment]
Hey, maybe a little late, but in 4.6 there is actually a method of the `Msh` class called `to_multiblock`  which will split different tissues into separate blocks (MultiBlock format) and also conserve the node and cell data
```
Convert a Msh instance to PyVista MultiBlock format (similar to
vtkMultiBlockDataSet). The result can be saved as a .vtm file

            mb = mesh.to_multiblock()
            mb.save("mesh.vtm")

and viewed in Paraview.
```
This allows you to toggle different tissues on and off. 

https://github.com/simnibs/simnibs/blob/7059ef0b96a65fb02e2a84652ace14585f60c9fe/simnibs/mesh_tools/mesh_io.py#L3718
